src/main/scala/scalaz/parsers/core/types.scala

```diff
@@ -1,8 +1,10 @@
-package scalaz
+package scalaz.parsers.core
 
+import scalaz.parsers.alt.{ AlternativeClass, ApplicativeErrorClass, MonadErrorClass }
+import scalaz.parsers.defs.ProductFunctorClass
 import scalaz.tc.InstanceOf
 
-package object parsers {
+object types {
```

Comment:
I kept them at package level to avoid extra import at use-site. These types are normally defined in scalaz and the only need to have them here is due to our scalaz-8 dependency. Having no import at use-site allows simpler support for scalaz-7.